astropy/coordinates/earth.py

```diff
@@ -236,9 +236,9 @@ def from_geocentric(cls, x, y, z, unit=None):
         if unit is None:
             try:
                 unit = x.unit
-            except AttributeError:
+            except AttributeError as err:
```

Comment:
```suggestion
            except AttributeError:
```

Since the subsequent TypeError is now "from None" 👍